core/server/master/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -203,14 +205,10 @@ public String getName() {
   }
 
   @Override
-  public void processJournalCheckpoint(JournalInputStream inputStream) throws IOException {
-    // clear state before processing checkpoint.
-    mBlocks.clear();
-    super.processJournalCheckpoint(inputStream);
-  }
-
-  @Override
   public void processJournalEntry(JournalEntry entry) throws IOException {
+    if (entry.getSequenceNumber() == 0) {
+      mBlocks.clear();
```

Comment:
Can you leave a comment that this is the first entry of checkpoint, so you want to clear the state.